ouroboros-consensus/src/Ouroboros/Consensus/Protocol/Abstract.hs

```diff
@@ -165,10 +165,10 @@ class ( Show (ChainState    p)
   -- | Apply a block
   --
   -- TODO this will only be used with headers
```

Comment:
This can be removed.
Same comment as above: is this only meant for headers or do we also use it with blocks sometimes?